src/suites/cts/validation/setBindGroup.spec.ts

```diff
@@ -0,0 +1,193 @@
+
+    const pipeline = this.device.createComputePipeline({
+      computeStage: { module, entryPoint: 'main' },
+      layout: pipelineLayout,
+    });
+
+    const encoder = this.device.createCommandEncoder();
+    const computePass = encoder.beginComputePass();
+    computePass.setPipeline(pipeline);
+    computePass.setBindGroup(0, bindGroup, dynamicOffsets);
+    computePass.dispatch(1, 1, 1);
+    computePass.endPass();
+
+    encoder.finish();
+  }
+
+  testRenderPass(dynamicOffsets: number[]): void {
+    const { bindGroup, pipelineLayout } = this.setup();
+
+    const vertexModule = this.device.createShaderModule({
+      code: GLSL(
+        'vertex',
+        `#version 450
+          void main() {
+            gl_Position = vec4(0);
+          }
+        `
+      ),
+    });
+
+    const fragmentModule = this.device.createShaderModule({
+      code: GLSL(
+        'fragment',
+        `#version 450
+          layout(std140, set = 0, binding = 0) uniform UniformBuffer {
+              vec2 value1;
+          };
+          layout(std140, set = 0, binding = 1) buffer StorageBuffer {
+              vec2 value2;
+          };
+          void main() {
+          }
+        `
+      ),
+    });
+
+    const pipeline = this.device.createRenderPipeline({
+      vertexStage: { module: vertexModule, entryPoint: 'main' },
+      fragmentStage: { module: fragmentModule, entryPoint: 'main' },
+      layout: pipelineLayout,
+      primitiveTopology: 'triangle-list',
+      colorStates: [{ format: 'rgba8unorm' }],
+    });
+
+    const attachmentTexture = this.device.createTexture({
+      format: 'rgba8unorm',
+      size: { width: 16, height: 16, depth: 1 },
+      usage: GPUTextureUsage.OUTPUT_ATTACHMENT,
+    });
+
+    const encoder = this.device.createCommandEncoder();
+    const renderPass = encoder.beginRenderPass({
+      colorAttachments: [
+        {
+          attachment: attachmentTexture.createView(),
+          loadValue: { r: 1.0, g: 0.0, b: 0.0, a: 1.0 },
+        },
+      ],
+    });
+    renderPass.setPipeline(pipeline);
+    renderPass.setBindGroup(0, bindGroup, dynamicOffsets);
+    renderPass.draw(3, 1, 0, 0);
+    renderPass.endPass();
+
+    encoder.finish();
+  }
+}
+
+export const g = new TestGroup(F);
+
+g.test('dynamic offsets match expectations in pass encoder', async t => {
+  const { dynamicOffsets, success } = t.params;
+
+  if (success) {
+    // Control case
+    t.testComputePass(dynamicOffsets);
+    t.testRenderPass(dynamicOffsets);
+  } else {
+    // Dynamic offsets don't match expectations in compute pass.
+    await t.expectValidationError(() => {
+      t.testComputePass(dynamicOffsets);
+    });
+    // Dynamic offsets don't match expectations in render pass.
+    await t.expectValidationError(() => {
+      t.testRenderPass(dynamicOffsets);
+    });
+  }
+}).params([
+  { dynamicOffsets: [256, 0], success: true }, // Dynamic offsets aligned
+  { dynamicOffsets: [], success: false }, // No dynamic offsets
```

Comment:
Can we have a test with only one dynamic offset, and with 3?

Also a test with a bindgroup with no dynamic offsets and give a non-empty list.